packages/cli/src/commands/alter.test.ts

```diff
@@ -0,0 +1,205 @@
+    };
+    jest.spyOn(mockedFallBackRegistry, 'publish').mockResolvedValue([]);
+    jest.spyOn(mockedFallBackRegistry, 'getMetaUrl').mockResolvedValue(metaUrl);
+    jest.spyOn(CannonStorage.prototype, 'readDeploy').mockResolvedValue(testPkgData);
+    jest.spyOn(CannonStorage.prototype, 'putDeploy').mockResolvedValue(newUrl);
+    jest.mocked(createDefaultReadRegistry).mockResolvedValue(Promise.resolve(mockedFallBackRegistry));
+    jest.mocked(getMainLoader).mockReturnValueOnce({
+      file: localLoader,
+      ipfs: ipfsLoader,
+    });
+  });
+
+  test('should perform alteration for set-url command', async () => {
+    // Set up test data and variables
+    const command = 'set-url';
+    const targets = [url];
+
+    const newTestPkgData: DeploymentInfo = _.assign(testPkgData, {
+      miscUrl: 'file:/usecannon.com/new-misc',
+    });
+
+    const mockReadDeploy = jest.fn(async (url) => {
+      switch (url) {
+        case 'file:/usecannon.com/misc':
+          return { misc: 'info' };
+        case 'file:/usecannon.com/meta':
+          return { itsMeta: 'data' };
+        case 'file:/usecannon.com':
+          return testPkgData;
+        case url:
+          return testPkgData;
+        case newUrl:
+          return newTestPkgData;
+      }
+    });
+
+    jest.spyOn(localLoader, 'read').mockImplementation(mockReadDeploy);
+
+    // Call the 'alter' function with the necessary arguments
+    await alter(packageName, chainId, preset, testPkgData.meta, command, targets, runtimeOverrides);
+
+    expect(CannonStorage.prototype.readDeploy as jest.Mock<any, any>).toHaveBeenCalledWith(packageName, preset, chainId);
+    expect(CannonStorage.prototype.putDeploy as jest.Mock<any, any>).toHaveBeenCalledWith(newTestPkgData);
+    expect(mockedFallBackRegistry.publish as jest.Mock<any, any>).toHaveBeenCalledWith(
+      [packageName],
+      `${chainId}-${preset}`,
+      newUrl,
+      metaUrl
+    );
+  });
+
+  test('should perform alteration for set-contract-address command', async () => {
+    // Set up test data and variables
+    const command = 'set-contract-address';
+    const targets = ['TestContract', '0x2222222222222222222222222222222222222222'];
+
+    // Call the 'alter' function with the necessary arguments
+    await alter(packageName, chainId, preset, testPkgData.meta, command, targets, runtimeOverrides);
+
+    expect(CannonStorage.prototype.readDeploy as jest.Mock<any, any>).toHaveBeenCalledWith(packageName, preset, chainId);
+    expect(CannonStorage.prototype.putDeploy as jest.Mock<any, any>).toHaveBeenCalledWith(testPkgData);
+    expect(testPkgData.state['provision.dummyStep'].artifacts.contracts!.TestContract.address).toEqual(targets[1]);
+    expect(mockedFallBackRegistry.publish as jest.Mock<any, any>).toHaveBeenCalledWith(
+      [packageName],
+      `${chainId}-${preset}`,
+      newUrl,
+      metaUrl
+    );
+  });
+
+  // TODO
+  // it('should perform alteration for mark-complete', async () => {});
```

Comment:
@dbeal-eth it looks like the package status must be modified, what do you think?